src/jit/morph.cpp

```diff
@@ -3312,15 +3303,58 @@ GenTreeCall* Compiler::fgMorphArgs(GenTreeCall* callNode)
+                            bool needCpyBlk = false;
+                            if (lclVar != NULL)
+                            {
+                                // If the struct is promoted to registers, it has to be materialized
+                                // on stack. We may want to support promoted structures in
+                                // codegening pugarg_stk instead of creating a copy here.
+                                LclVarDsc*  varDsc = &lvaTable[lclVar->gtLclVarCommon.gtLclNum];
+                                needCpyBlk = varDsc->lvPromoted;
+                            }
+                            else 
+                            {
+                                // The following pattern needs copy to store the
+                                // result of the simd operation. Otherwise, this                               
+                                // pattern reaches codegen that doesn't expect
+                                // addr node.
+                                // Eventually, rationalizer can handle this 
+                                // instead of generating copy here.
+                                //                                
+                                // +--*  obj       simd16
+                                // |  \--*  addr      byref
+                                // |     |  /--*  lclVar    simd16 V05 loc4
+                                // |     \--*  simd      simd16 int -
+                                // |        \--*  lclVar    simd16 V08 tmp1
+                                //
+                                GenTree* addr = argObj->gtOp.gtOp1;
+                                if (addr->OperGet() == GT_ADDR) 
+                                {
+                                    GenTree* addrChild = addr->gtOp.gtOp1;
+                                    if (addrChild->OperGet() == GT_SIMD) 
+                                    {
+                                        needCpyBlk = true;
+                                    }                                    
+                                }
+                            }
                             passStructInRegisters = false;
-                            copyBlkClass = NO_CLASS_HANDLE;
+                            if (needCpyBlk)
+                            {
+                                copyBlkClass = objClass;
+                            }
+                            else
+                            {                                
+                                copyBlkClass = NO_CLASS_HANDLE;
+                            }
                         }
                         else
                         {
                             // The objClass is used to materialize the struct on stack.
                             passStructInRegisters = true;
                             copyBlkClass = objClass;
                         }
+                        
```

Comment:
For case 1-5, if `eeGetSystemVAmd64PassStructInRegisterDescriptor` classifies them as register argument, it calls `fgMakeOutgoingStructArgCopy` and copies are generated. For case 6, we eliminated some copies. I will open an issue to avoid copies for case 1-5.